api/client/commands.go

```diff
@@ -2183,7 +2183,7 @@ func (cli *DockerCli) createContainer(config *runconfig.Config, hostConfig *runc
 	//create the container
 	stream, statusCode, err := cli.call("POST", "/containers/create?"+containerValues.Encode(), mergedConfig, false)
 	//if image not found try to pull it
-	if statusCode == 404 {
+	if statusCode == 404 && strings.Contains(err.Error(), config.Image) {
```

Comment:
Yeah--we definitely need to make sure that we only go into that block if it really is "image not found" on that specific image, so I think your test is probably the only way to be exactly sure, although I hate the string matching--so brittle :-/  But, I know this has come up before--and we don't have much choice unless we get typed errors all the way through the codebase.